src/simplay-jupyter/src/index.ts

```diff
@@ -29,37 +36,186 @@ export class RenderSimplay extends Widget implements IRenderMime.IRenderer {
+      );
+      const currentStepInfo = document.createElement('p');
+      currentStepInfo.classList.add('simplay-label');
+      simulationSpooler.addStepChangedEventListener((ts) => {
+        currentStepInfo.innerText = 'Current Step: ' + ts;
+      });
+      const startPauseButton = this.createStartPauseButton(simulationSpooler);
+
+      const resetButton = this.createResetButton();
+      resetButton.addEventListener('click', () => {
+        startPauseButton.reset();
+        simulationSpooler.reset();
+      });
+
+      const advanceOneStepButton = this.createAdvanceOneStepButton();
+      advanceOneStepButton.addEventListener('click', () => {
+        simulationSpooler.advanceOneStep();
+      });
+
+      const spacer = this.createSpacer();
+
+      const speedInputName = 'skipTo';
+      const speedInputLabel = this.createLabel('speedInputLabelId', speedInputName, 'Speed: ');
+      const speedInput = this.createSpeedInput(speedInputName);
+      speedInput.addEventListener('change', (event: Event) => {
+        if (event.currentTarget) {
+          simulationSpooler.setSpeedFactor(
+            Number((event.currentTarget as HTMLInputElement).value)
+          );
+        }
+      });
+
+      const skipToInputName = 'skipTo';
+      const skipToLabel = this.createLabel('skipToLabelId', skipToInputName, 'Skip to: ');
+      const skipToInput = this.createSkipToInput(skipToInputName);
+      skipToInput.addEventListener('keyup', (event: Event) => {
+        if (event.currentTarget) {
+          simulationSpooler.skipTo(
+            Number((event.currentTarget as HTMLInputElement).value)
+          );
+        }
+      });
+
+      const spacer2 = this.createSpacer();
+
+      // Changing the order of the append calls affects the actual order in the UI.
+      controls.appendChild(startPauseButton.button);
+      controls.appendChild(resetButton);
+      controls.appendChild(advanceOneStepButton);
+      controls.appendChild(spacer);
+      controls.appendChild(speedInputLabel);
+      controls.appendChild(speedInput);
+      controls.appendChild(skipToLabel);
+      controls.appendChild(skipToInput);
+      controls.appendChild(currentStepInfo);
+      controls.appendChild(spacer2);
     });
 
-    this.node.appendChild(controls);
+    // Start observing the target node for configured mutations
+    observer.observe(simplayContainer);
+
+    simplayContainer.appendChild(controls);
     return Promise.resolve();
   }
 
+  private createSkipToInput(name: string) {
+    const skipToInput = document.createElement('input');
+    skipToInput.type = 'number';
+    skipToInput.name = name;
+    skipToInput.classList.add('simplay-number-input');
+    return skipToInput
+  }
+
+  private createSpeedInput(name: string) {
+    const speedInput = document.createElement('input');
+    speedInput.type = 'range';
+    speedInput.min = '0.2';
+    speedInput.max = '20';
+    speedInput.step = '0.2';
+    speedInput.value = '1';
```

Comment:
We could think about making these constants, but this is nit picking on my side :)